src/cache/impl/LevelKVStorage.ts

```diff
@@ -0,0 +1,71 @@
+import { Level } from 'level';
+import { MemoryLevel } from 'memory-level';
+import { DB } from '@ethereumjs/trie';
+import { CacheOptions } from '../../core/WarpFactory';
+import { LoggerFactory } from '../../logging/LoggerFactory';
+import { BatchDBOp } from '@ethereumjs/trie/dist/types';
+
+const ENCODING_OPTS = { keyEncoding: 'buffer', valueEncoding: 'buffer' };
+
+export class LevelKVStorage implements DB {
+  private readonly logger = LoggerFactory.INST.create('LevelKVStorage');
+  private _db: MemoryLevel;
+
+  constructor(private readonly cacheOptions: CacheOptions) {}
+
+  // Lazy initialization upon first access
+  private get db(): MemoryLevel {
+    if (!this._db) {
+      if (this.cacheOptions.inMemory) {
+        this._db = new MemoryLevel(ENCODING_OPTS);
+      } else {
+        if (!this.cacheOptions.dbLocation) {
+          throw new Error('LevelDb cache configuration error - no db location specified');
+        }
+        const dbLocation = this.cacheOptions.dbLocation;
+        this.logger.info(`Using location ${dbLocation}`);
+        this._db = new Level<string, any>(dbLocation, ENCODING_OPTS);
+      }
+    }
+    return this._db;
+  }
+
+  async get(key: Buffer): Promise<Buffer | null> {
+    let value = null;
+    try {
+      await this.db.open();
+      value = await this.db.get(key, ENCODING_OPTS);
+    } catch (error) {
+      // This should be `true` if the error came from LevelDB
+      // so we can check for `NOT true` to identify any non-404 errors
+      if (error.notFound !== true) {
+        throw error;
+      }
+    } finally {
+      await this.db.close();
+    }
+    return value;
+  }
+
+  async put(key: Buffer, val: Buffer): Promise<void> {
+    await this.db.open();
```

Comment:
without manual open and close I'm getting "Database is not open" errors. 

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/84311757/208767713-6831cd57-7d3e-4c84-9d1d-24ad600c4abb.png">